apis/core/v1alpha1/featureflagconfiguration_types.go

```diff
@@ -38,15 +41,56 @@ type FeatureFlagConfigurationSpec struct {
 	// +optional
 	// +nullable
 	FlagDSpec *FlagDSpec `json:"flagDSpec"`
-	// FeatureFlagSpec is the json representation of the feature flag
-	FeatureFlagSpec string `json:"featureFlagSpec,omitempty"`
+	// FeatureFlagSpec is the json representation of the feature flag specification
+	// Deprecated: use FeatureFlagSpecV2
+	FeatureFlagSpec *string `json:"featureFlagSpec,omitempty"`
+	// FeatureFlagSpec is the structured representation of the feature flag specification
+	FeatureFlagSpecV2 *FeatureFlagSpec `json:"featureFlagSpecV2,omitempty"`
+}
+
+func (ffcs FeatureFlagConfigurationSpec) FeatureFlagSpecJSON() (string, error) {
+	var ffcsJson string
+	if ffcs.FeatureFlagSpecV2 != nil { // prioritise V2
+		ffcsJsonB, err := json.Marshal(ffcs.FeatureFlagSpecV2)
+		if err != nil {
+			return "", fmt.Errorf("FeatureFlagSpecV2: %w", err)
+		}
+
+		ffcsJson = string(ffcsJsonB)
+	} else if ffcs.FeatureFlagSpec != nil {
+		ffcsJson = *ffcs.FeatureFlagSpec
+	} else {
+		return "", errors.New("FeatureFlagSpecV2 and FeatureFlagSpec are empty")
+	}
+
+	return ffcsJson, nil
 }
 
 type FlagDSpec struct {
 	// +optional
 	Envs []corev1.EnvVar `json:"envs"`
 }
 
+type FeatureFlagSpec struct {
+	Flags map[string]FlagSpec `json:"flags"`
+}
+
+type FlagSpec struct {
+	// +kubebuilder:validation:Enum=ENABLED;DISABLED
+	State string `json:"state"`
```

Comment:
Semantically we'd probably like this to be an limited set of values (`ENABLED`/`DISABLED`) but I think last I checked it was difficult to unmarshall JSON into enums, so it might not be possible/worth it.

Reply:
Oh, I see the `kubebuilder` at least validates it, so that seems good enough.